syncserver/src/tokenserver/extractors.rs

```diff
@@ -209,12 +181,8 @@ impl FromRequest for TokenserverRequest {
             log_items_mutator.insert("metrics_uid".to_owned(), hashed_fxa_uid.clone());
 
             // To preserve anonymity, compute a hash of the FxA device ID to be used for reporting
-            // metrics. Only requests using BrowserID will have a device ID, so use "none" as
-            // a placeholder for OAuth requests.
-            let hashed_device_id = {
-                let device_id = auth_data.device_id.as_deref().unwrap_or("none");
```

Comment:
I think we should kill the `AuthData::device_id` field entirely now as well (now always `None` and no longer used?)

Reply:
Yeah, I didn't want to touch that because it could get complicated, fast. The problem is that `device_id` gets converted to `hashed_device_id`, which has a lot of dependencies, most notably, as part of the HAWK header request. I'll remove the `device_id`, but I'm also going to file a ticket to see about removing `hashed_device_id`, because I think we may need to check the HAWK header bits to make sure that there's no weirdness that could happen if that field goes away.